packages/components/src/Disclosure/Disclosure.tsx

```diff
@@ -27,35 +24,43 @@ interface DisclosureProps {
    * @default false
    */
   readonly defaultOpen?: boolean;
+
+  /**
+   * Callback that is called when the disclosure is toggled.
+   */
+  readonly onToggle?: (newOpened: boolean) => void;
+
+  /**
+   * Used to make the disclosure a Controlled Component.
+   */
+  readonly open?: boolean;
 }
 
 export function Disclosure({
   children,
   title,
   defaultOpen = false,
+  onToggle,
+  open,
 }: DisclosureProps) {
-  const [isOpen, setOpen] = useState(defaultOpen);
-  const [isMounted, setMount] = useState(false);
-  const [titleRef, { exactWidth }] = useResizeObserver<HTMLDivElement>();
-  const isBelowBreakpoint = exactWidth && exactWidth < Breakpoints.small;
+  const [internalOpen, setInternalOpen] = useState(
+    defaultOpen || open || false,
+  );
+  const isOpen = open !== undefined ? open : internalOpen;
+  const { smallAndUp } = useBreakpoints();
   const isTitleString = typeof title === "string";
 
-  useEffect(() => {
-    setMount(true);
-  }, []);
-
   return (
-    <details open={isOpen} onToggle={onToggle} className={styles.details}>
-      <summary className={styles.summary}>
+    <details open={isOpen} className={styles.details}>
+      <summary className={styles.summary} onClick={handleToggle}>
```

Comment:
Use `onClick` here instead of `onToggle` to prevent `details` not working as controlled disclosure. For more information [see](https://github.com/facebook/react/issues/15486#issuecomment-488028431)